test-case/check-s3.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+
+##
+## Case Name: check S3 status
+## Preconditions:
+##    N/A
+## Description:
+##    Run the S3 command to check device status
+## Case step:
+##    1. switch S3 operation
+##    2. use rtcwake -m mem command to do S3
+##    3. check command return value
+##    4. check dmesg errors
+##    5. check wakeup increase
+## Expect result:
+##    S3 recover
+##    check kernel log and find no errors
+##
+
+source $(dirname ${BASH_SOURCE[0]})/../case-lib/lib.sh
+
+# wait time should >= 3 for other device wakeup from sleep
+random_min=3
+random_max=20
+
+OPT_OPT_lst['l']='loop'     OPT_DESC_lst['l']='loop count'
+OPT_PARM_lst['l']=1         OPT_VALUE_lst['l']=5
+
+OPT_OPT_lst['p']='power'    OPT_DESC_lst['p']="S3 power type from /sys/power/mem_sleep"
```

Comment:
s2idle is used for S0ix in Linux.
For s3 it is using` s2idle [deep]`

Reply:
so make the deep as the default option?

Reply:
@Bin-QA My suggestion is to keep the default value and output them.
@aiChaoSONG will those GLK have default s2idle setting?

Reply:
@xiulipan , Due to the coreboot issue, S3 is not supported on GLK, we need to bypass S3 test on GLK.

Reply:
@keqiaozhang one thing to confirm, will GLK setting default to be `s2idle `or `deep`?

Reply:
Default value is deep on GLK.